src/OrleansProviders/Streams/Common/PooledCache/IObjectPool.cs

```diff
@@ -41,6 +47,11 @@ public abstract class PooledResource<T> : IDisposable
         public IObjectPool<T> Pool { set { pool = value; } }
 
         /// <summary>
+        /// Pooled resource size
+        /// </summary>
+        public virtual int SizeInByte { get {return Marshal.SizeOf(this);} }
```

Comment:
this is the default implementation, which is here just for not returning null. PooledResource's child class which we are actually using in cache system all override this method to a much cheaper method,